packages/@react-spectrum/label/test/StaticField.test.js

```diff
@@ -0,0 +1,328 @@
+    );
+
+    let staticField = getByTestId('test-id');
+    expect(staticField).toBeInTheDocument();
+    expect(staticField).toHaveTextContent('wow, cool, and awesome');
+  });
+
+  it('renders correctly with CalendarDate value', function () {
+    let {getByTestId} = render(
+      <StaticField
+        data-testid="test-id"
+        label="Field label"
+        value={new CalendarDate(2019, 6, 5)} />
+    );
+
+    let staticField = getByTestId('test-id');
+    expect(staticField).toBeInTheDocument();
+    expect(staticField).toHaveTextContent('6/5/2019');
+  });
+
+  it('renders correctly with CalendarDate value with user provided format options', function () {
+    let {getByTestId} = render(
+      <StaticField
+        data-testid="test-id"
+        label="Field label"
+        value={new CalendarDate(2019, 6, 5)}
+        formatOptions={{dateStyle: 'long'}} />
+    );
+
+    let staticField = getByTestId('test-id');
+    expect(staticField).toBeInTheDocument();
+    expect(staticField).toHaveTextContent('June 5, 2019');
+  });
+  
+
+  it('renders correctly with CalendarDateTime value', function () {
+    let {getByTestId} = render(
+      <StaticField
+        data-testid="test-id"
+        label="Field label"
+        value={new CalendarDateTime(2020, 2, 3, 12, 23, 24, 120)} />
+    );
+
+    let staticField = getByTestId('test-id');
+    expect(staticField).toBeInTheDocument();
+    expect(staticField).toHaveTextContent('2/3/2020');
+  });
+
+  it('renders correctly with CalendarDateTime value with user provided format options', function () {
+    let {getByTestId} = render(
+      <StaticField
+        data-testid="test-id"
+        label="Field label"
+        value={new CalendarDateTime(2020, 2, 3, 12, 23, 24, 120)}
+        formatOptions={{dateStyle: 'medium', timeStyle: 'medium'}} />
+    );
+
+    let staticField = getByTestId('test-id');
+    expect(staticField).toBeInTheDocument();
+    expect(staticField).toHaveTextContent('Feb 3, 2020, 12:23:24 PM');
+  });
+
+  it('renders correctly with ZonedDateTime value', function () {
+    let {getByTestId} = render(
+      <StaticField
+        data-testid="test-id"
+        label="Field label"
+        value={new ZonedDateTime(2020, 2, 3, 'America/Los_Angeles', -28800000)} />
+    );
+
+    let staticField = getByTestId('test-id');
+    expect(staticField).toBeInTheDocument();
+    expect(staticField).toHaveTextContent('2/3/2020');
+  });
+
+  it('renders correctly with ZonedDateTime value with user provided format options', function () {
+    let {getByTestId} = render(
+      <StaticField
+        data-testid="test-id"
+        label="Field label"
+        value={new ZonedDateTime(2020, 3, 3, 'America/Los_Angeles', -28800000)}
+        formatOptions={{dateStyle: 'full', timeStyle: 'short'}} />
+    );
+
+    let staticField = getByTestId('test-id');
+    expect(staticField).toBeInTheDocument();
+    expect(staticField).toHaveTextContent('Tuesday, March 3, 2020 at 3:00 AM');
```

Comment:
Currently looking into why this fails on circleci, looks like the formatted date time is dependent on the local timezone of the user running the test which is what we expect since this is ZonedDateTime not CalendarDateTime. The assertion will have to be flexible to accommodate this behavior, thinking of how to do so short of grabbing the local time zone and doing a separate time format option to produce the same string

Reply:
You can use https://jestjs.io/docs/expect#tomatchregexp--string to match the start of the string and ignore the time

Reply:
hm, can't think of an alternative than basically rehashing the logic present in StaticField to generate the expected result:
```
    let formatOptions = {dateStyle: 'full', timeStyle: 'short'};
    let zoned = new ZonedDateTime(2020, 3, 3, 'America/Los_Angeles', -28800000);
    let formatter = new DateFormatter('en-US', formatOptions);
    let expected =  formatter.format(zoned.toDate());
    let {getByTestId} = render(
      <StaticField
        data-testid="test-id"
        label="Field label"
        value={zoned}
        formatOptions={{dateStyle: 'full', timeStyle: 'short'}} />
    );

    let staticField = getByTestId('test-id');
    expect(staticField).toBeInTheDocument();
    expect(staticField).toHaveTextContent(expected);
```